backend/plugin/db/mysql/query.go

```diff
@@ -55,3 +59,42 @@ func getStatementWithResultLimitForTiDB(singleStatement string, limitCount int)
 	}
 	return "", nil
 }
+
+// singleStatemtn must be a selectStatement for mysql.
+func getStatementWithResultLimitForMySQL(singleStatement string, limitCount int) (string, error) {
+	list, err := mysqlparser.ParseMySQL(singleStatement)
+	if err != nil {
+		return "", err
+	}
+
+	listener := &mysqlRewriter{
+		limitCount: limitCount,
+	}
+
+	for _, stmt := range list {
+		listener.rewriter = *antlr.NewTokenStreamRewriter(stmt.Tokens)
+		antlr.ParseTreeWalkerDefault.Walk(listener, stmt.Tree)
+		if listener.err != nil {
+			return "", errors.Wrapf(listener.err, "statement: %s", singleStatement)
+		}
+	}
+	return listener.rewriter.GetTextDefault(), nil
+}
+
+type mysqlRewriter struct {
+	*mysql.BaseMySQLParserListener
+
+	rewriter   antlr.TokenStreamRewriter
+	err        error
+	limitCount int
+}
+
+func (r *mysqlRewriter) EnterQueryExpression(ctx *mysql.QueryExpressionContext) {
+	limitClause := ctx.LimitClause()
+	if limitClause != nil {
+		// limit clause already exists.
+		return
+	}
+
+	r.rewriter.InsertAfterDefault(ctx.GetStop().GetTokenIndex(), fmt.Sprintf(" LIMIT %d", r.limitCount))
+}
```

Comment:
It doesn't meet our expectations. It will change all `QueryExpression`'s `LIMIT`, such as subquery and CTE.

We need to only change the top `QueryExpression`.

Also, add tests for subquery, union and CTE